pool/app/settings/repo/repo.ml

```diff
@@ -119,6 +119,19 @@ module PageScripts = struct
     let find = find_opt tbl
     let add database_label = replace tbl database_label
     let update = add
+
+    let update_script label location script =
+      let cached_scripts = find label in
+      let new_scripts =
+        match cached_scripts, location with
+        | Some { body; _ }, Head -> { head = script; body }
+        | Some { head; _ }, Body -> { head; body = script }
+        | None, Head -> { head = script; body = None }
+        | None, Body -> { head = None; body = script }
```

Comment:
What happens for equal cases with cached script and location (e.g. `| Some { body; _ }, Body`) ?

The compiler seems to be ok, that these cases are missing, don't know why.